arlo_server/contests.py

```diff
@@ -110,13 +125,46 @@ def validate_contests(json_contests: List[dict]) -> None:
             )
 
 
+def round_status_by_contest(
+    round: Optional[Round], contests: List[Contest]
+) -> Dict[str, Optional[JSONDict]]:
+    if not round:
+        return {c.id: None for c in contests}
+
+    sampled_ballot_count_by_contest = dict(
+        SampledBallotDraw.query.filter_by(round_id=round.id)
+        .join(Batch)
+        .join(Jurisdiction)
+        .join(Jurisdiction.contests)
+        .group_by(Contest.id)
+        .values(Contest.id, func.count(SampledBallotDraw.batch_id))
+    )
+    round_is_complete_by_contest = dict(
+        RoundContest.query.filter_by(round_id=round.id).values(
+            RoundContest.contest_id, RoundContest.is_complete
+        )
+    )
+
+    # isRiskLimitMet will be None until we have computed the risk measurement
+    # for that contest, which happens once we're done auditing its sampled
+    # ballots. Once the risk measurement is calculated, isRiskLimitMet will be
+    # a boolean.
+    return {
+        c.id: {
+            "isRiskLimitMet": round_is_complete_by_contest[c.id],
+            "numBallotsSampled": sampled_ballot_count_by_contest.get(c.id, 0),
+        }
+        for c in contests
```

Comment:
wow I've never used dictionary comprehension, that's so cool :)

Reply:
Yeahhhh I'm so into the comprehensions!